crates/fuel-core/src/service/genesis.rs

```diff
@@ -69,33 +71,89 @@ pub fn maybe_initialize_state(
 ) -> anyhow::Result<()> {
     // check if chain is initialized
     if database.ids_of_latest_block()?.is_none() {
-        import_genesis_block(config, database)?;
+        import_chain_state(config, database)?;
+        commit_genesis_block(config, database)?;
     }
 
     Ok(())
 }
 
-fn import_genesis_block(
+fn import_chain_state(
     config: &Config,
     original_database: &Database,
 ) -> anyhow::Result<()> {
-    // start a db transaction for bulk-writing
-    let mut database_transaction = Transactional::transaction(original_database);
+    let block_height = config.chain_config.height.unwrap_or_default();
+
+    let coins_reader = config.state_decoder.coins()?;
+    let mut _coin_roots =
+        import_coin_configs(&original_database, coins_reader, block_height)?;
+
+    _coin_roots.sort();
+
+    // TODO: other threads should be killed if one encounters a failure
```

Comment:
Is there a follow-up item for this `TODO`?

Reply:
These TODOs in `genesis.rs` are addressed in the follow up PR